File: 120. Triangle/120. Triangle.cs
public class Solution {
    public int MinimumTotal(IList<IList<int>> triangle) {
        int[] dp = triangle[triangle.Count - 1].ToArray();
        
        for (int i = triangle.Count - 2; i >= 0; i--) {
            for (int j = 0; j <= i; j++) {
                dp[j] = triangle[i][j] + Math.Min(dp[j], dp[j + 1]);
            }
        }
        
        return dp[0];
    }
}

File: 120. Triangle/120. Triangle.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int[] dp = new int[triangle.get(triangle.size() - 1).size()];
        for (int j = 0; j < dp.length; j++) {
            dp[j] = triangle.get(triangle.size() - 1).get(j);
        }
        
        for (int i = triangle.size() - 2; i >= 0; i--) {
            for (int j = 0; j <= i; j++) {
                dp[j] = triangle.get(i).get(j) + Math.min(dp[j], dp[j + 1]);
            }
        }
        
        return dp[0];
    }
}